SGA-160-Improve-friends-search

GillWell opened the pull request:
<!-- Enter a brief description/summary of your PR here. What does it fix/what does it change/how was it tested (even manually, if necessary)? -->
## Summary of the Pull Request

<!-- Please review the items on the PR checklist before submitting-->
## PR Checklist

- [x] Add searching by username
- [x] Add hinting when users start typing email or username
- [x] Start hinting when user types in at least 3 letters

<!-- Provide a more detailed description of the PR, other things fixed or any additional comments/features here -->
## Detailed Description of the Pull Request / Additional comments

- Added search field to look for users by their names

- Each search field will search field will show hints after 3 symbols are written

- Hints will not show your own email and name

- If you press on a hint of a user that you are already friends with it will show a message

GillWell committed:
Improvments

- Added search field to look for users by their names

- Each search field will search field will show hints after 3 symbols are written

- Hints will not show your own email and name

- If you press on a hint of a user that you are already friends with it will show a message

paulius-mongirdas commented on app/src/main/java/com/github/k409/fitflow/ui/screen/friends/FriendRequestScreen.kt:
```diff
@@ -100,7 +112,81 @@ fun FriendRequestContent(
+                onClick = {
+                    coroutineScope.launch {
+                        foundUser = viewModel.searchUserByEmail(searchText.lowercase())
+                        if (foundUser?.uid == "") {
+                            SnackbarManager.showMessage(context.getString(R.string.user_not_found))
+                        }
```
There is an issue where snackbar messages are hidden behind the keyboard. You could try to resolve this by adding `android:windowSoftInputMode="adjustResize"` in the manifest file.

paulius-mongirdas commented on app/src/main/java/com/github/k409/fitflow/ui/screen/friends/FriendRequestScreen.kt:
```diff
@@ -133,6 +219,35 @@ fun FriendRequestContent(
         }
+
+        if (searchTextByName.length >= 3 && searchResultsByName.isNotEmpty()) {
+            Column(
+                modifier = Modifier
+                    .padding(vertical = 8.dp)
+                    .padding(start = 18.dp, end = 12.dp),
+            ) {
+                searchResultsByName.forEach { user ->
```
There might be an issue where if there are hundreds of search results, then the whole screen will get filled up with search results and user would need to scroll down a lot to add a friend. Also, search results do not go away if user selects an email or name. 
You could try to resolve this by modifying your existing code or rework searching input fields using Material 3 SearchBar composable, which solves most of these problems.
https://m3.material.io/components/search/overview

utkarsh006 replied:
@GillWell you can take reference of [this code](https://github.com/utkarsh006/Drinks/blob/main/app/src/main/java/com/example/hoichi/presentation/drink_list_screen/components/SearchComponent.kt) for searchbar

GillWell committed:
Search bar changes

- added [ android:windowSoftInputMode="adjustResize" ] for the snackbar problem

- made it so that only 5 hints at most would be shown at any time of writing a user email or name

GillWell committed:
Merge branch 'dev' into SGA-160-Improve-friends-search

mantasjasikenas committed:
Remove duplicated localeConfig value in manifest
